v1.10.0 Baselines and Alert Runners as Snowflake TASK's

sfc-gh-afedorov opened the pull request:
# Data Connectors

- Adds NIC's to AzIC
- Adds `s3.get_public_access_block` to AWSIC
- Adds GreatHorn event ingestion
- AWSIC has group and user inline policies collection added by @stevenliu-czi in #550 (ty!)

# Baselines

- Basic Baseline implemented in TASKS + SQL, removing deprecated R runner
- Basic Baseline UI on by default

# Docker-based runners

- Sentry for error logging

# Runners

- Adds retry to `db.fetch` in alert processor to catch exception

# Handlers
- Add SNS handler region envar (thank you, @ugmuka)

sfc-gh-afedorov committed:
Removes baselines runners & rpy

Commit:
Enable Sentry exception tracking

sfc-gh-afedorov committed:
[pack] Adds ingest/jamf_inventory

whiteops-github committed:
Update authentication.rst

sfc-gh-afedorov committed:
Update mypy.ini

sfc-gh-afedorov committed:
Update Dockerfile.snowalert

sfc-gh-afedorov committed:
Update CONTRIBUTING.md

sfc-gh-afedorov committed:
Update azure_collect.py

sfc-gh-pkommini committed:
Add azic nics (#447)

sfc-gh-afedorov committed:
Update connectors_runner.py

sfc-gh-afedorov committed:
Fix CCI tests (#450)

sfc-gh-pkommini committed:
Add custom event_time column as an option during baseline creation (#448)

sfc-gh-afedorov committed:
Add DC: GCP CloudAssets (#449)

* [dc] Adds gcp_cloudassets ingestion + pipeline

* feedback from PK

sfc-gh-pkommini committed:
Add active connections list tab to the webUI (#451)

sfc-gh-afedorov committed:
Fix Docker build

sfc-gh-afedorov committed:
Update gcp_cloudassets.py

sfc-gh-afedorov committed:
Update gcp_cloudassets.py

sfc-gh-afedorov committed:
Update Dockerfile.webui

Commit:
[DC] Add aws_collect s3.get_public_access_block (#454)

sfc-gh-pkommini committed:
Fix aws collect ut (#456)

* Fix aws_collect unit test

* Fix aws_collect unit test

sfc-gh-afedorov committed:
Add Accountwide S3 Bucket Public Access Block to AWSIC (#459)

Co-authored-by: edulop <elopez@chanzuckerberg.com>

sfc-gh-afedorov committed:
Run new version of Prettify on all ts, tsx (#462)

sfc-gh-afedorov committed:
Update Dockerfile.webui

reduce likelihood of data loading timeouts

sfc-gh-pkommini committed:
Fix prettier lint-all and run prettier on all directories. (#463)

sfc-gh-pkommini committed:
Fix lint issues with prettier (#465)

Commit:
update aws_query_pack to match GRANT statement (#461)

* Match GRANT SELECT statement to created rule above it to match
convention

sfc-gh-pkommini committed:
Add format button on RawEditor (#460)

* Add format button on RawEditor

* Fix yarn.lock

sfc-gh-pkommini committed:
add PR template (#466)

* add pr template

* Lint fixes

* Lint fixes

* Lint fixes

sfc-gh-pkommini committed:
Fix format button to activate if code is not empty (#467)

* Fix format button to activate if code is not empty

* Update RawEditor.tsx

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>

sfc-gh-afedorov committed:
Update menu.ts

sfc-gh-mpolyakova committed:
Greathorn logs (#428)

Co-authored-by: sfc-gh-mpolyakova <maria.polyakova@snowflake.com>

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Update setup.py

sfc-gh-afedorov committed:
Update greathorn_logs.py

apparently utcnow is no longer kosher https://stackoverflow.com/a/2175170

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Update data-views.sql.fmt

zoltan-fedor committed:
Adding handler for MS Teams (#471)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Adding handler to MS Teams

This allows us to send alerts to MS Teams channels via webhooks.

If you don't know how to get the webhook for an MS channel:
https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using#setting-up-a-custom-incoming-webhook

A typical handler using this new 'msteams' handler looks like the following:
```
      ARRAY_CONSTRUCT(
        OBJECT_CONSTRUCT(
           'type', 'msteams',
           'webhook', 'https://outlook.office.com/webhook/xxxxxx/IncomingWebhook/xxxxx/xxxxx',
           'title', 'Resource created in Snowflake.',
           'color', '#FF0000',
           'message', 'This is a test alert; the actor is ' || actor || ' and the action is ' || action || '.'
        )
      )
```

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>

sfc-gh-pkommini committed:
Change dependency to snowsql-formatter (#469)

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Update jamf_inventory.sql

Commit:
[DC] Add S3 Get Bucket Tagging (#476)

* [DC] Add S3 Get Bucket Tagging

* fix the order

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
Clean up tests

sfc-gh-afedorov committed:
Fix test builds (#478)

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-pkommini committed:
Add h1 data connector (#475)

sfc-gh-ttruong committed:
Delete h1_collect.png

sfc-gh-ttruong committed:
Replace with better fit H1 logo

sfc-gh-afedorov committed:
Fix Jira link sort order

When searching for similar tickets finding the most recent is better than the first

sfc-gh-dwyleczukstern committed:
add ability to collect nat gateways (#482)

* Add ability to collect nat gateways
* Update migrations/v1_9_6-v1_10_0.md

sfc-gh-afedorov committed:
Update setup.py

sfc-gh-afedorov committed:
Merge branch 'v1.10.0' of github.com:snowflakedb/SnowAlert into v1.10.0

sfc-gh-dwyleczukstern committed:
Update v1_9_6-v1_10_0.md

Add in network interface details

sfc-gh-dwyleczukstern committed:
Update v1_9_6-v1_10_0.md

sfc-gh-dwyleczukstern committed:
add describe network interfaces (#484)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* add describe network interfaces

* fix field, add outpost

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-afedorov committed:
Update README.md

sfc-gh-ttruong committed:
Add pagination h1 collect (#483)

* Modularize features into separate functions
* update page size
* fix defaults for data and amount values

sfc-gh-afedorov committed:
Create CODEOWNERS

sfc-gh-afedorov committed:
Update setup.py

sfc-gh-afedorov committed:
Merge branch 'v1.10.0' of github.com:snowflakedb/SnowAlert into v1.10.0

sfc-gh-afedorov committed:
update python dep's

sfc-gh-dwyleczukstern committed:
[DC] Add EFS Collection to AWSIC (#485)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* add efs collection

* Update aws_collect.py

Typo fix

* fix missing param, minor bugs

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: Andrey Fedorov <andrey.fedorov@snowflake.com>

sfc-gh-ttruong committed:
Fix amount and transactions (#486)

- Fix the null values for award amount

sfc-gh-afedorov committed:
Update h1_collect.py

ran it through pyblack formatter

sfc-gh-ttruong committed:
H1 fix transactions (#487)

* update date in transactions table

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-afedorov committed:
Fix versions, remove mypy

sfc-gh-afedorov committed:
Fix typo in SMTP handler

sfc-gh-afedorov committed:
Generic ExtFunc lambda + examples (#432)

sfc-gh-afedorov committed:
Update greathorn_logs.py

sfc-gh-afedorov committed:
[EF] add bearer auth, minor fixes

- add bearer auth
- add UA required by GH API
- support JSON in headers
- add execute-api:Invoke grant to TF

sfc-gh-afedorov committed:
Update snowalert.tf

sfc-gh-afedorov committed:
add unit to CloudWatch EF

sfc-gh-afedorov committed:
fix EF header names in packs

sfc-gh-afedorov committed:
add region name to TF

sfc-gh-pkommini committed:
Add sort active conns (#479)

sfc-gh-afedorov committed:
Google Collect Connector (#490)

* Create google_collect.py

* Update google_collect.py

sfc-gh-pkommini committed:
Add groups and members query (#492)

sfc-gh-afedorov committed:
Update send_mail.sql.fmt

sfc-gh-afedorov committed:
Add verbosity option to https req

sfc-gh-afedorov committed:
[processor] Fix correlation logic on lists (#495)

sfc-gh-afedorov committed:
Update alert_processor.py

sfc-gh-dwyleczukstern committed:
collect more GCP data (#494)

* collect more GCP data

* update prefix

sfc-gh-afedorov committed:
[ef] Improve error message

return expected responses to Snowflake rather than to CloudWatch logs

sfc-gh-afedorov committed:
[ef tests] update zip test

sfc-gh-afedorov committed:
Update https.py

sfc-gh-afedorov committed:
[ef] refactor and add secretsmanager, xmlrpc, boto3 support

sfc-gh-afedorov committed:
Update utils.py

sfc-gh-afedorov committed:
[dc] Add SCC Findings to GCPIC (#515)

sfc-gh-afedorov committed:
Update google_collect.py

sfc-gh-afedorov committed:
[ef] Add Google processor

sfc-gh-afedorov committed:
Update violation_queries_runner.py (#521)

sfc-gh-dwyleczukstern committed:
increase limit when populating users in an okta group (#473)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* increase limit when populating users in an okta group

* handled pagination

* moved next url to method as requested

* Update okta.py

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>

sfc-gh-dwyleczukstern committed:
Revert "increase limit when populating users in an okta group (#473)" (#524)

This reverts commit 2c07ea4c8329edb85a95983dfadb7f0562fe8df8.

sfc-gh-afedorov committed:
Fix run.py to pass args to connectors, various cleanup (#525)

* [mypy] fix support for latest mypy

* remove old docs

* Bump pyyaml from 5.3.1 to 5.4 in /src (#518)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Add .whitesource configuration file (#498)

Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>

* Bump pyyaml from 5.3.1 to 5.4 in /src

Bumps [pyyaml](https://github.com/yaml/pyyaml) from 5.3.1 to 5.4.
- [Release notes](https://github.com/yaml/pyyaml/releases)
- [Changelog](https://github.com/yaml/pyyaml/blob/master/CHANGES)
- [Commits](https://github.com/yaml/pyyaml/compare/5.3.1...5.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Update dependency axios to ^0.21.0 (#513)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Add .whitesource configuration file (#498)

Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>

* Update dependency axios to ^0.21.0

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>
Co-authored-by: WhiteSource for GitHub.com <whitesource-for-github-com[bot]@users.noreply.github.com>

* Bump cryptography from 3.0.0 to 3.2 in /src (#455)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Bump cryptography from 3.0.0 to 3.2 in /src

Bumps [cryptography](https://github.com/pyca/cryptography) from 3.0.0 to 3.2.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/3.0...3.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump y18n from 4.0.0 to 4.0.1 in /src/webui/frontend (#522)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Add .whitesource configuration file (#498)

Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>

* Bump y18n from 4.0.0 to 4.0.1 in /src/webui/frontend

Bumps [y18n](https://github.com/yargs/y18n) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/yargs/y18n/releases)
- [Changelog](https://github.com/yargs/y18n/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yargs/y18n/commits)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump elliptic from 6.5.3 to 6.5.4 in /src/webui/frontend (#496)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Bump elliptic from 6.5.3 to 6.5.4 in /src/webui/frontend

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.3 to 6.5.4.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.3...v6.5.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump axios from 0.20.0 to 0.21.1 in /src/webui/frontend (#480)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Bump axios from 0.20.0 to 0.21.1 in /src/webui/frontend

Bumps [axios](https://github.com/axios/axios) from 0.20.0 to 0.21.1.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v0.21.1/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.20.0...v0.21.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump ini from 1.3.5 to 1.3.7 in /src/webui/frontend (#472)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Bump ini from 1.3.5 to 1.3.7 in /src/webui/frontend

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* [cci] remove build-docs job

* Update yarn.lock

* Update run.py

* remove details from active connections

* Update yarn.lock

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>
Co-authored-by: WhiteSource for GitHub.com <whitesource-for-github-com[bot]@users.noreply.github.com>

sfc-gh-afedorov committed:
[dc] Group by account in AWS IC

sfc-gh-afedorov committed:
Revert "[dc] Group by account in AWS IC"

This reverts commit f9720753961f83ce21c43a86a38695e295b163ff.

sfc-gh-afedorov committed:
[dc] Add iam_get_account_authorization_details to AWSIC

sfc-gh-afedorov committed:
fix formatting

sfc-gh-afedorov committed:
[ef] rename "processors" to "drivers"

sfc-gh-afedorov committed:
Add Jira list values in custom fields (#526)

sfc-gh-afedorov committed:
[dc] fix AWSIC Session Cache

sfc-gh-afedorov committed:
[dc] fix AWSIC session cache

sfc-gh-afedorov committed:
[dc] add debug flag to connector runner

sfc-gh-afedorov committed:
Update lambda_function.py

sfc-gh-gkadam committed:
[DC] Add EC2 Describe Snapshots to AWSIC (#527)

sfc-gh-afedorov committed:
[dc] Change AWSIC to 6 retries

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-afedorov committed:
Add Categories to Alert Queries, Jira (#477)

sfc-gh-pkommini committed:
Modify https driver (#532)

sfc-gh-afedorov committed:
Revert "[dc] Change AWSIC to 6 retries"

This reverts commit 10d649fdff7aa0c829d608bd662c7b9c053439d8.

sfc-gh-afedorov committed:
[dc] Fix AWSIC AioConfig timeout params

sfc-gh-afedorov committed:
[dc] Fix AWSIC formatting

sfc-gh-afedorov committed:
[dc] add data schema envar to AWSIC

sfc-gh-afedorov committed:
[dc] awsic get credentials report after loading users

sfc-gh-afedorov committed:
[dc] refresh awsic session between regions

sfc-gh-afedorov committed:
[geff] add authorized-user-info option to google driver

tested w/ SP on calendar API

sfc-gh-afedorov committed:
Update process_google.py

sfc-gh-afedorov committed:
[awsic] Session by boto client

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-afedorov committed:
Update mypy.ini

sfc-gh-afedorov committed:
[dispatcher] optimize updating of alerts table (#537)

sfc-gh-afedorov committed:
[handler] Optimize Jira ticket updates

sfc-gh-afedorov committed:
Optimize alert_id ref's in ASR, processor, helper views

sfc-gh-afedorov committed:
[aqr] skip merge when no query results

sfc-gh-afedorov committed:
[slack] raise errors rather than logging

sfc-gh-afedorov committed:
[webui] Update some dependencies

sfc-gh-afedorov committed:
[webui] bump version number

sfc-gh-rmahajan commented:
Added the code to v1.10.0 as suggested by Prasanth. @anfedorov can you please review this.

sfc-gh-rmahajan committed:
Slack Group Message (#544)

* Slack Group Message

* Slack messages for Group

* Formatting the code

sfc-gh-afedorov committed:
Revert "Slack Group Message (#544)"

This reverts commit 6bd4f9357bd3d5e500dd4ac1fcabe1126c85460d.

sfc-gh-rmahajan committed:
Add Slack group IM option (#545)

sfc-gh-afedorov committed:
Fix new mypy errors

sfc-gh-afedorov committed:
Revert "Fix new mypy errors"

This reverts commit de0f3c34f1d26ca1ef0e915dfd1d22e687e29d25.

sfc-gh-afedorov committed:
[mypy] fix new mypy dep's install, new errors (#547)

sfc-gh-afedorov committed:
[AQR] Create TEMP tables instead of TRANSIENT

sfc-gh-afedorov committed:
Revert "[AQR] Create TEMP tables instead of TRANSIENT"

This reverts commit 9bc259bbe7ad37032c628cdf534c2074cd51f2cc.

sfc-gh-afedorov committed:
Update alert_dispatcher.py (#548)

sfc-gh-afedorov committed:
[dc] Add raw col to AzIC Users

the users resource has a new "@odata.id" column which adding a raw column helps capture. tested locally.

stevenliu-czi committed:
[DC] Collect AWSIC IAM managed/inline policy data for users/groups/roles (#550)

sfc-gh-afedorov committed:
Remove R runner code

sfc-gh-afedorov committed:
Return query id from AQR error exceptions

sfc-gh-afedorov committed:
Fix violations run order

sfc-gh-afedorov committed:
Add zone support to aws_collect

sfc-gh-afedorov committed:
[awsic] Fix bug when providing multiple oid's

There was a bug when multiple oid's were provided

sfc-gh-afedorov committed:
[awsic] Add aws_zone option, fix defaults

sfc-gh-dwyleczukstern committed:
[awsic] add azure collect public IP addresses (#571)

* add azure collect public IP addresses
* Add migration, test, fix missing col's
Co-authored-by: Andrey Fedorov <andrey.fedorov@snowflake.com>

sfc-gh-afedorov committed:
[fix] Fix parsing of "undefined" in VARIANT results (#578)

Snowflake sometimes returns an "undefined" value inside of arrays created by SQL expression CREATE_ARRAY(NULL)

sfc-gh-afedorov committed:
Update Python dependencies to build on new install (#580)

* Fix dependencies on new install
- new version of pip requires demjson3
- new M1 requires new version crypto
- snowflake-connector-python needs update

sfc-gh-afedorov committed:
Update python deps (#581)

sfc-gh-afedorov committed:
Fix broken demjson reference (#584)

sfc-gh-afedorov committed:
Update JS dependencies (#582)

* Update JS dependencies

* Delete .babelrc

* Fix remaining vulns: nth-check, css-what and follow-redirects (#586)

* Yarn resolution for nth-check and css-what

* Bump follow-redirects from 1.14.7 to 1.14.8

* Make resolution path specific

* Update yarn.lock

Co-authored-by: Nachiket Lele <77716642+sfc-gh-nlele@users.noreply.github.com>

sfc-gh-nlele committed:
Add retry to db.fetch in alert_processor.py (#597)

sfc-gh-afedorov committed:
[dc] Fix AWSIC after aioboto3 update

sfc-gh-afedorov committed:
[dc] Revert aioboto3 update and fix

This reverts commit a52b43a7fc2d896f6a93c3f67447d26ef480c440.

and part of commit 36b6c4e8ebf6d691bce95b8154f75f371e8769b7

sfc-gh-afedorov committed:
[dispatcher] Ignore all handlers with "-" in type

sfc-gh-afedorov committed:
[dispatcher] Completely ignore alerts with dashes in handler type

this prevents us from un-handling handled alerts from UDF dispatchers

sfc-gh-nlele committed:
Enable mypy on untyped definitions (#596)

* Add flag to mypy for type checking definitions

* Test pytest

* Update config.yml

* Fix mypy errors

* Update test.py

* Create test.py

* Fix whitespaces

Co-authored-by: Andrey <42752788+sfc-gh-afedorov@users.noreply.github.com>

ugmuka committed:
Add sns handler region

fixes #606

sfc-gh-afedorov committed:
Update aws_collect.py

sfc-gh-bkou committed:
GSE-2738: Add ENTITIES and TAGS column to SnowAlert alerting (#641)

sfc-gh-kpatil committed:
GSE-3598 (#660)

* WIP: changing auth to bearer token based auth

Co-authored-by: Andrey <andrey.fedorov@snowflake.com>

sfc-gh-afedorov committed:
Add managed_by column to AzIC NICs (#662)

* Fix header date when already parsed

* Add managed_by column to AzIC NICs table

sfc-gh-afedorov committed:
Add raw column to azure_collect_pricings (#663)

sfc-gh-afedorov committed:
GSE-3635: Fix AWSIC logging and status=408 errors (#664)

* Remove vestigial logging

no need to log because these errors are recorded into the "error" column in the corresponding table

* Add retry and update aio* dependencies

Update aiohttp, aioboto3, and aiobotocore dependencies and add retry logic to sts.assume_role invocation.

sfc-gh-afedorov committed:
Update base image to Python 3.8 on Debian bookworm (#665)

* Update base image to Debian bookworm

* Update to py38

sfc-gh-gtan committed:
Revert "Update base image to Python 3.8 on Debian bookworm (#665)" (#667)

This reverts commit a66a8176171c96b4a6849a5c8002fd6f9cfbe1df.

sfc-gh-gtan committed:
Revert "GSE-3635: Fix AWSIC logging and status=408 errors (#664)" (#666)

This reverts commit a80713c6f7f28dd79d70c122f1b752dee6f0d395.

sfc-gh-gtan committed:
Update README.md (#669)

sfc-gh-afedorov committed:
Update Python and pip dependencies & fix tests (#670)

* Update Python and pip dependencies & fix tests

* fix share db view creation in installer script

Snowflake added a new column so the share db name is col4 now

sfc-gh-gtan committed:
[GSE-4024] Tag image with commit SHA after build, and push to repository (#671)

* Tag image with commit SHA after build, and push to repository

* updates to contibution file

sfc-gh-gtan committed:
missing bash (#672)

sfc-gh-afedorov committed:
Reapply "GSE-3635: Fix AWSIC logging and status=408 errors (#664)" (#666) (#673)

This reverts commit b7b512682924fefe00faf442f01debc03a07c3f5.

sfc-gh-afedorov committed:
Update Dockerfile.snowalert (#674)

fixing error
> botocore.exceptions.CredentialRetrievalError: Error when retrieving credentials from container-role: Error retrieving metadata: Received error when attempting to retrieve container metadata: Connect timeout on endpoint URL: "http://<ip>/v2/credentials/<uuid>"

per https://repost.aws/questions/QUCFqv7OfoQlygJrmwfkJ24Q/various-aws-apis-fail-due-to-timeout

sfc-gh-afedorov committed:
Update Dockerfile.snowalert

sfc-gh-afedorov committed:
Bump aioboto3 to 13.0.1

sfc-gh-afedorov committed:
Remove iptables Docker installation

sfc-gh-afedorov committed:
Move to non-slim Debian

sfc-gh-afedorov committed:
Add 5m sleep to Docker file

sfc-gh-afedorov committed:
Revert "Add 5m sleep to Docker file"

This reverts commit b84705b247a0c4e57ada9d6f65081f15cb2c35b0.

sfc-gh-afedorov committed:
add retry on CredentialRetrievalError

sfc-gh-afedorov committed:
Revert "Move to non-slim Debian"

This reverts commit 6c0d2bd53a62230eb027941d4064a9fee5fa17d7.

sfc-gh-afedorov committed:
fix retry logic and add log

sfc-gh-afedorov committed:
Update utils.py

sfc-gh-afedorov committed:
Update utils.py

sfc-gh-afedorov committed:
Update utils.py

sfc-gh-gtan committed:
[GSE-4150] Update azure_collect.py to fix `KeyError: 'managedByExtended'` error (#675)

* Update azure_collect.py to fix `KeyError: 'managedByExtended'` error

* Update v1_9_6-v1_10_0.md

sfc-gh-afedorov committed:
Reduce AWSIC ingestion speed

getting throttled at metadata service returning header "X-Rate-Limit-Limit: 40.00"

sfc-gh-afedorov committed:
Update AWS Collect to reduce batch size

sfc-gh-afedorov committed:
Fix AWS Inventory & Configuration Collection (#676)

* fix awsic rate limiting and OOM errors

- rate limit metadata separately
- prune cache to fix oom errors
- remove / abstract out inline retries

* optimizations

- use boto's retry mechanism for 429 responses
- cache AssumeRole failures indefinitely
- re-use sessions between regions
- refactor API rate limits as AioRateLimit instances
- increase batch size to 1000

* add pause in pagination

sfc-gh-afedorov committed:
Move down storage_account_containers

Putting them last because they currently time out after a few days of ingestion

sfc-gh-pkommini committed:
fix reduce the insert batch size (#687)

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* revert formatting changes.

* revert formatting changes.

* add comment explaining batch size

---------

Co-authored-by: Andrey <andrey.fedorov@snowflake.com>

sfc-gh-nlele committed:
Update setup.py (#688)

sfc-gh-pkommini committed:
fix: Reduce the insert batch size (#689)

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* Reduce the insert batch size to reduce compilation memory usage.

* revert formatting changes.

* revert formatting changes.

* add comment explaining batch size

* Reduce page size further.

---------

Co-authored-by: Andrey <andrey.fedorov@snowflake.com>

sfc-gh-bkou committed:
Patch SnowAlert container image vulnerabilities. (#691)

sfc-gh-pkommini committed:
fix(connectors): guard per-call/page failures so one error doesn't abort collection (#692)

* Update Dockerfile.snowalert

* Update Dockerfile.snowalert

* Add .whitesource configuration file (#498)

Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>

* Update service_now.py

* revert aded63b Update service_now.py

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* fix(aws_collect): handle account- and region-level outages without aborting collection

Catch BotoCoreError and ClientConnectionError at the region, account, and
coroutine levels so that a single unreachable AWS region or account no longer
crashes the entire collection batch for all remaining resources.

.... Generated with [Cortex Code](https://docs.snowflake.com/en/user-guide/cortex-code/cortex-code)

Co-Authored-By: Cortex Code <noreply@snowflake.com>

* fix(connectors): guard per-call failures in azure, crowdstrike, okta, tenable_io

Wraps individual API calls and page fetches in try/except so a single
credential, endpoint, or network error skips that item and continues
collection instead of aborting the entire batch — consistent with the
aws_collect fix.

.... Generated with [Cortex Code](https://docs.snowflake.com/en/user-guide/cortex-code/cortex-code)

Co-Authored-By: Cortex Code <noreply@snowflake.com>

* refactor(connectors): tighten exception types and trim comments

- aws_collect: use isinstance(e, ClientError) instead of hasattr(e, 'response')
  for safer access to the response attribute when recording account-level errors
- okta: narrow bare Exception catches to requests.exceptions.RequestException
  in ingest_users and the logs pagination loop
- trim verbose inline comments to one-liners across all five connectors

.... Generated with [Cortex Code](https://docs.snowflake.com/en/user-guide/cortex-code/cortex-code)

Co-Authored-By: Cortex Code <noreply@snowflake.com>

* revert(connectors): remove per-call error guards, let exceptions propagate

Reverts the try/except wrapping added in aws_collect, azure_collect,
crowdstrike_devices, okta, and tenable_io that silently swallowed
per-call, per-page, and per-scanner failures. Exceptions now propagate
so failures are visible rather than silently skipped.

.... Generated with [Cortex Code](https://docs.snowflake.com/en/user-guide/cortex-code/cortex-code)

Co-Authored-By: Cortex Code <noreply@snowflake.com>

---------

Co-authored-by: sfc-gh-afedorov <42752788+sfc-gh-afedorov@users.noreply.github.com>
Co-authored-by: whitesource-for-github-com[bot] <50673670+whitesource-for-github-com[bot]@users.noreply.github.com>
Co-authored-by: Andrey <andrey.fedorov@snowflake.com>
Co-authored-by: Cortex Code <noreply@snowflake.com>

sfc-gh-afedorov committed:
SADC migrate AWSIC findings to Inspector2 (#694)

* migrate AWSIC findings to Inspector2

* limit Inspector2 to enabled AWS regions

* log skipped AWS regions

* restore AWS connection timeout default

sfc-gh-afedorov committed:
handle AWS connect timeouts (#695)

sfc-gh-afedorov committed:
filter AWS regions by opt-in status (#696)

* filter AWS regions by opt-in status

* surface AWS connection failures

* do not persist AWS network errors
